.github/workflows/config.yml

```diff
@@ -0,0 +1,139 @@
+name: Configuration
+
+on:
+  workflow_call:
+    outputs:
+      # manually defined values
+      autotest_data_s3_url:
+        description:
+        value: "s3://data-autotest/test_data_2024-nov-07"
+      vcpkg_version:
+        description: "vcpkg version"
+        value: "2024.10.21" # VCPKG-AUTO-UPDATE - refer 'vcpkg-auto-update.yml' for more info
+      # automatically computed or tag-based values
+      app_version:
+        description: "Version without namespace: v1.2.3.4"
+        value: ${{ jobs.prepare-config.outputs.app_version }}
+      full_config_build:
+        description:
+        value: ${{ github.event_name == 'schedule' || jobs.prepare-config.outputs.tag-full-ci == 'true' }}
+      docker_image_tag:
+        description:
+        value: ${{ jobs.prepare-config.outputs.docker-image-tag }}
+      internal_build:
+        description:
+        value: ${{ github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name == github.repository }}
+      need_linux_image_rebuild:
+        description:
+        value: ${{ jobs.prepare-config.outputs.linux-changes == 'true' && github.event_name != 'schedule' && jobs.prepare-config.outputs.tag-skip-image-rebuild != 'true' }}
+      need_windows_vcpkg_rebuild:
+        description:
+        value: ${{ jobs.prepare-config.outputs.windows-changes == 'true' || jobs.prepare-config.outputs.tag-bump-vcpkg == 'true' }}
+      release_tag:
+        description: "Version with namespace (if it exists): v1.2.3.4-namespace"
+        value: ${{ jobs.prepare-config.outputs.release_tag }}
+      update_doc:
+        description:
+        value: ${{ jobs.prepare-config.outputs.tag-update-doc == 'true' }}
+      upload_artifacts:
+        description:
+        value: ${{ github.event_name == 'push' || jobs.prepare-config.outputs.tag-full-ci == 'true' }}
+      upload_test_artifacts:
+        description:
+        value: ${{ jobs.prepare-config.outputs.tag-upload-test-artifacts == 'true' }}
+
+jobs:
+  prepare-config:
```

Comment:
mb we can mv `set-ubuntu-x64-matrix` from `build-test-ubuntu-x64.yml` to this `prepare-config`?